src/main/java/duchess/Deadline.java

```diff
@@ -0,0 +1,48 @@
+package duchess;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents a task with a deadline. 
+ * A {@code Deadline} object stores the task name and the due date/time.
+ */
+public class Deadline extends Task{
+    LocalDateTime by;
```

Comment:
`by` can be set to private since it's not used anywhere else and can be kept encapsulated.
```suggestion
    private LocalDateTime by;
```